bolt/runtime/sys_riscv64.h

```diff
@@ -0,0 +1,94 @@
+#ifndef LLVM_TOOLS_LLVM_BOLT_SYS_RISCV64
+#define LLVM_TOOLS_LLVM_BOLT_SYS_RISCV64
+
+#include <dirent.h>
```

Comment:
When cross-compiling, these headers come from the sysroot (not from the host) so this shouldn't be an issue. Or am I missing something?

By the way, I have been cross-compiling this while developing instrumentation. This is my workflow:
```
$ riscv64-linux-gnu-g++ -c -ffreestanding -fno-exceptions -fno-rtti -fno-stack-protector -fPIC $LLVM_SRC/bolt/runtime/instr.cpp -o instr.o -I $LLVM_BUILD/tools/bolt/bolt_rt-bins/
$ ar -crs $LLVM_BUILD/lib/libbolt_rt_instr-riscv64.a instr.o
$ $LLVM_BUILD/bin/llvm-bolt --instrument --runtime-instrumentation-lib=libbolt_rt_instr-riscv64.a ...
```

Reply:
The runtime files should match the system your input binaries were compiled for. Because BOLT is independent of the compiler, historically we never assumed that you actually have the full target toolchain, so the few target constants we depend on are hardcoded. But we can probably change to include the kernel header files directly and then make sure our cmake rules correctly configure the compiler used to build this runtime lib.

Reply:
I was reading our current cmake file and we explicitly disable building the runtime if llvm is being cross-compiled. I imagine we would need to update the build rules to better support building the runtime for different scenarios.

Reply:
As I mentioned on another PR, the clean solution would probably be to move to `runtimes` build model (sanitizer, instrumentation runtimes etc). This should enable proper native and cross- compilation. But since we're not there yet, I'd stick with hard-coded definitions for syscall numbers.

Reply:
> But since we're not there yet, I'd stick with hard-coded definitions for syscall numbers.

I feel like I'm missing the problem we're trying to solve here. It seems to me this would only be useful in a situation where the user does have a (cross-) compiler available for the target (otherwise they couldn't compile the instrumentation library) but somehow does not have those system headers available. Is this a realistic scenario? I don't think I have ever seen a toolchain in the wild that doesn't ship those headers.

I also just noticed that `runtime/common.h` includes `cstddef` and `cstdint` so we are already relying on the toolchain to provide certain headers.

Reply:
Ping.